File: Python_Solutions/CodeWars/7kyu/Simple_string_characters.py
# In this Kata, you will be given a string and your task will be to return a list of ints detailing the count of uppercase letters, lowercase, numbers and special characters, as follows.
#
# Solve("*'&ABCDabcde12345") = [4,5,5,3].
# --the order is: uppercase letters, lowercase, numbers and special characters.
# More examples in the test cases.
#
# Good luck!
#
# FUNDAMENTALS
# Solution
def solve(s):
    uc, lc, num, sp = 0, 0, 0, 0
    for i in s:
        if i.isupper(): uc += 1
        elif i.islower(): lc += 1
        elif i.isdigit(): num += 1
        else: sp += 1
    return [uc, lc, num, sp]

File: CodeWars/7kyu/Simple_string_characters.cpp
/*
In this Kata, you will be given a string and your task will be to return a list of ints detailing the count of uppercase letters, lowercase, numbers and special characters (everything else), as follows.

The order is: uppercase letters, lowercase letters, numbers and special characters.

"*'&ABCDabcde12345" --> [ 4, 5, 5, 3 ]
More examples in the test cases.

Good luck!

FUNDAMENTALS
*/
// Solution
std::vector<int> solve(std::string s){
    std::vector<int> output(4, 0);
    for (char lt : s) {
      if (lt >= 'A' && lt <= 'Z') {
        output[0] += 1;
      } else if (lt >= 'a' && lt <= 'z') {
        output[1] += 1;
      } else if (lt >= '0' && lt <= '9') {
        output[2] += 1;
      } else {
        output[3] += 1;
      }
    }
  return output;
}